tests/test_update_query.py

```diff
@@ -117,6 +117,34 @@ def test_with_query_list_int():
     assert str(url.with_query([("a", 1)])) == "http://example.com/?a=1"
 
 
+@pytest.mark.parametrize(
+    "query,expected",
+    [
+        pytest.param({"a": [1, 2]}, "?a=1&a=2", id="list"),
+        pytest.param({"a": (1, 2)}, "?a=1&a=2", id="tuple"),
+        pytest.param({"a": ["1", 2]}, "?a=1&a=2", id="mixed types"),
+        pytest.param({"a": 1, "b": [2, 3]}, "?a=1&b=2&b=3", id="single then list",),
+        pytest.param({"a": [1, 2], "b": 3}, "?a=1&a=2&b=3", id="list then single",),
+    ],
+)
+def test_with_query_sequence(query, expected):
+    url = URL("http://example.com")
+    assert url.with_query(query) == URL("http://example.com/" + expected)
+
+
+@pytest.mark.parametrize(
+    "query",
+    [
+        pytest.param({"a": [[1]]}, id="nested"),
+        pytest.param([("a", [1, 2])], id="tuple list"),
+    ],
+)
+def test_with_query_sequence_invalid_use(query):
+    url = URL("http://example.com")
+    with pytest.raises(TypeError):
```

Comment:
the best practice is to always use `match=` arg
```suggestion
    with pytest.raises(TypeError):
```

Reply:
None of the other existing tests do this, so I tried to stay in line with them (and also was unaware of such thing). Made a commit with your suggestion.